Cb 8 zip code api

Aarushi-Pandey opened the pull request:
* **Please check if the PR fulfills these requirements**
- [x] The pull request title contains the ticket number in the title. For example [CB-1]


* **What kind of change does this PR introduce?** (Bug fix, new feature...)
calls zipcode API using a supplied zip code


* **What is the current behavior?What does the code do before you made the new changes?** 
Currently, the application does not call the zipcode api.


* **What is the new behavior?**
The application calls the zipcode api and returns a json of the data from localhost8080.


* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR? Run a new command?)
Add the weather api key to the .env file


* **Other information**:

Aarushi-Pandey committed:
created component to ask user for zipcode and display it

Aarushi-Pandey committed:
added input for distance user wants to travel

Aarushi-Pandey committed:
added other components to Landing page

Aarushi-Pandey committed:
Made a new frontend component for User Input, added routing, unable to fix styling error in Landing Page

Aarushi-Pandey committed:
called zipcode api in localhost 8080, unable to pass user zipcode and distance from the user input component

Aarushi-Pandey committed:
successfully made zipcode api call, unable to call api from the zipcode component

Aarushi-Pandey committed:
successful api call from index.js

Codubee commented:
Hey Arushi. I think there are some changes in here that are not part of the zip code api

JerryTeng01 commented on backend/index.js:
```diff
@@ -6,12 +6,26 @@ app.use(express.json())
 
 app.listen(port,()=>{
-    console.log('API is up and running')
+    console.log('Zipcode API is up and running')
```
change this since zip code api is not the only api running

JerryTeng01 commented on backend/index.js:
```diff
@@ -6,12 +6,26 @@ app.use(express.json())
+const distance = 50;
+
+app.get('/api/zipcodes',(req,res)=>{
+
```
you can get the parameters from req.query instead of hardcoding them

JerryTeng01 commented on backend/index.js:
```diff
@@ -6,12 +6,26 @@ app.use(express.json())
+
+app.get('/api/zipcodes',(req,res)=>{
+
+    const headers = {headers: {authorization: process.env.API_KEY}}
```
change the name of the API key to something that describes what it is since there will be multiple api keys

JerryTeng01 replied:
does this API actually require a key? It doesnt seem like it does

JerryTeng01 reviewed (commented):
Seems good, just make sure to include the changes from main

Codubee commented on frontend/src/components/UserInput/Zipcode/Zipcode.js:
```diff
@@ -0,0 +1,65 @@
+import React from 'react'
```
Is this file part of this pr?

Codubee commented on frontend/src/components/UserInput/Zipcode/Zipcode.css:
```diff
@@ -0,0 +1,21 @@
+.container {
```
Is this file part of this pr?

Codubee commented on frontend/src/components/UserInput/UserInput.js:
```diff
@@ -0,0 +1,12 @@
+import React from 'react'
```
Is this file part of this pr?

Codubee commented on frontend/src/App.js:
```diff
@@ -1,5 +1,6 @@
 import './App.css';
```
Is this file part of this pr?

Codubee commented on frontend/package.json:
```diff
@@ -3,6 +3,7 @@
   "version": "0.1.0",
```
Is this file part of this pr?

Codubee reviewed (changes requested):
There are some files for the frontend when I dont think there should be. Might consider removing them from this pr

Aarushi-Pandey committed:
included recent changes from main, got parameters for zipcode api from req.query instead of hardcoding them, removed api key since it is not needed for the zipcode api to be called, removed unrelated files from this pr

Aarushi-Pandey commented:
Thanks so much for the detailed feedback Miguel and Jerry! It really helped me improve my code efficiently and quickly. I've updated this branch so that there are no merge conflicts with main and have changed my hardcoded zipcode and distance values in index.js to query parameters as Jerry suggested. Great job in figuring out that there was no need for the API key for the zipcode API as well! I've removed that as well. I've attempted to remove frontend files from this pr too. Please let me know if there are further changes to be made! Thanks once more for taking the time to give me detailed comments and suggestions :)

Codubee commented on backend/index.js:
```diff
@@ -6,6 +6,24 @@ app.use(express.json());
+    console.log(zipcode);
+    console.log(distance);
+
+    axios.get('https://www.zipcodeapi.com/rest/radius.json/'+zipcode+'/'+distance+'/km')
```
Can we use miles instead of kilometers

Codubee reviewed (changes requested):
Fantastic work fixing everything! Just left one comment and then ill approve.

Aarushi-Pandey committed:
changed distance from kilometers to miles when calling zipcode api

Aarushi-Pandey commented:
Changed distance from kilometers to miles.
